### PR TITLE
Tweak README instruction about running development version

### PR DESCRIPTION
**Pull request by SobhanMP**

### CODE REVIEW

**Comment by SobhanMP** [comment]
ok so turns out i did not understand PKG_GIT... it's pointless for deps that aren't in Julialang/julia

**Comment by SobhanMP** [comment]
@ViralBShah

**Comment by fredrikekre** [comment]
Why doesn't `DEPS_GIT="SparseArrays"` work?

**Comment by SobhanMP** [comment]
because it's checkouts the SHA1 commit set in SparseArrays.version. i tested it 3 times

ref:
https://github.com/JuliaLang/julia/blob/db570dfb8d7607d98e71d092a2933a44bf4e3f55/deps/tools/git-external.mk#L51

**Comment by fredrikekre** [comment]
Yes, as noted in the Pkg note https://github.com/JuliaLang/Pkg.jl/blob/master/README.md#using-the-development-version-of-pkgjl, but it gives you a git repo which you can do whatever you want with, i.e. checkout master if you like.

**Comment by SobhanMP** [comment]
but what's the point? it's going to give you a git repo no matter what?

**Comment by SobhanMP** [comment]
ok i just saw your message on slack.

i guess it would be in the line of

```
make DEPS_GIT="SparseArrays" -C deps getall
```
checkout whatever commit manually in stdlib/SparseArrays
then
```
make DEPS_GIT="SparseArrays"
```

**Comment by ViralBShah** [comment]
@fredrikekre What's the recommendation here for the README?

**Comment by SobhanMP** [comment]
ok the instructions should be clear and correct.

File: README.md [hunk 1]
```diff
@@ -22,11 +22,11 @@ SparseArrays.jl provides functionality for working with sparse arrays in Julia.
 
 ## Using newer version of this package with julia master
 
-You need to build Julia from scratch to use the git version (or other version) of this package. The build process is the same, but `DEPS_GIT` should be set when building i.e. `make DEPS_GIT="SparseArrays" ...`. The other option is to manually select the commit in `stdlib/SparseArrays.version`. 
+To use a newer version of this package, you need to build Julia from scratch. The build process is the same as any other build except that `DEPS_GIT="SparseArrays"` should be passed to `make` i.e. `make DEPS_GIT="SparseArrays"`. Then after the build is complete, the git repo in `stdlib/SparseArrays` can be used to check out the desired commit/branch. Alternatively, you need to change the commit used in `stdlib/SparseArrays.version`. The first method does not require rebuilding julia to update the library.
```
Comment (fredrikekre) [inline README.md:25]: What do you mean with 
> The first method does not require rebuilding julia to update the library.

What is the first method referring to?

Comment (SobhanMP) [reply README.md:25]: using `DEPS_GIT`

Comment (fredrikekre) [reply README.md:25]: Why doesn't that require a rebuild?

Comment (SobhanMP) [reply README.md:25]: because if you can checkout the git repo.

Comment (SobhanMP) [reply README.md:25]: i guess there is no need to rebuild it in both cases but it's more convenient in the latter case? (tho i haven't tested it)

Comment (fredrikekre) [reply README.md:25]: Yea okay, but only because SparseArrays is not currently in the sysimage.

Comment (SobhanMP) [reply README.md:25]: should i add that detail?

Comment (SobhanMP) [reply README.md:25]: if it was in the sysimage, how would the git route look like?

Comment (fredrikekre) [reply README.md:25]: Same, but you would always have to rebuild.

Comment (ViralBShah) [reply README.md:25]: I think we can merge this for now - but if we put SparseArrays back in the sysimage, then we'll need to update.

**Comment by ViralBShah** [comment]
@dkarrasch Does this address your question about developing the package (unless I misunderstood it in another PR)?

**Comment by ViralBShah** [comment]
I'm merging this for now. The README can always be improved quickly if necessary.

**Pull request merged**